huxley/www/static/js/huxley/components/AdvisorRosterView.js

```diff
@@ -89,22 +127,61 @@ var AdvisorRosterView = React.createClass({
           <td>{delegate.email}</td>
           <td>{delegate.summary}</td>
           <td>
-            <Button>Delete</Button>
+            <Button color="red"
+                    size="small"
+                    onClick={this._handleDeleteDelegate.bind(this, delegate)}>
+              Delete
+            </Button>
           </td>
         </tr>
       )
     }.bind(this));
   },
 
-  // _handleAddDelegate: function() {
-  //   return (
-  //     <form>
-  //       <br>Name: <input type="text" placeholder="Name" valueLink={this.linkState('name')} /></br>
-  //       <br>Email: <input type="text" placeholder="Email" valueLink={this.linkState('email')}/></br>
-  //       <input type="submit" value="Submit" onclick={this._handleSubmit} />
-  //     </form>
-  //   )
-  // },
+  _handleAddDelegate: function() {
+    this.setState({adding_delegate: true});
+    return (
+      <Modal
+        isOpen={this.state.adding_delegate}
+        closeTimeoutMS={5}>
+        <h1>Modal Content</h1>
+        <p>Etc.</p>
+      </Modal>
+    );
+  },
+
+  openModal: function() {
+    this.setState({adding_delegate: true});
+  },
+
+  afterOpenModal: function() {
+    // references are now sync'd and can be accessed.
+    this.refs.subtitle.style.color = '#f00';
+  },
+
+  closeModal: function() {
+    this.setState({adding_delegate: false});
+  },
+
+  _handleDeleteDelegate: function(delegate) {
+    this.setState({loading: true});
+    $.ajax ({
+      type: 'DELETE',
+      url: '/api/delegates/'+delegate.id,
+      success: this._handleDelegateDeleteSuccess,
+      error: this._handleError,
+    });
+  },
+
+  _handleDelegateDeleteSuccess: function(data, status, jqXHR) {
+    var delegates = this.state.delegates
+    delegates = delegates.filter(function (delegate) {
+      return delegate.id != jqXHR.responseJSON.delegate.id
+    });
+    this.setState({delegates: delegates})
+    this.setState({loading: false});
+    this.history.pushState(null, '/advisor/roster');
```

Comment:
Will this cause the page to refresh and reload the delegates, or do we have to explicitly filter the delegates above like we do?